src/ship_cmd.cpp

```diff
@@ -624,7 +624,7 @@ bool IsShipDestinationTile(TileIndex tile, StationID station)
 	for (DiagDirection d = DIAGDIR_BEGIN; d != DIAGDIR_END; d++) {
 		TileIndex t = tile + TileOffsByDiagDir(d);
 		if (!IsValidTile(t)) continue;
-		if (IsDockTile(t) && GetStationIndex(t) == station && IsValidDockingDirectionForDock(t, d)) return true;
+		if (IsDockTile(t) && GetStationIndex(t) == station && IsDockWaterPart(t)) return true;
```

Comment:
If you plan to "merge with rebase", this commit message needs a bit of work. Such long sentences don't help anyone :P

``` 
Cleanup: docking tiles method names and parameters lost their meaning

This was caused by so many refactors, that it no longer makes sense.

```